examples/hat/binaryclock.py

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python
+import unicornhat as uh
+import datetime
+from time import sleep
+
+print("""Binar Clock made by Iorga Dragos Florian
+Displays Hour(red), Minute(yellow), Second(blue), Day(green) and Month(Pink) in binary format""")
+FPS = 30.0
+uh.rotation(0)
+uh.brightness(1)
+
+def draw_time(b,x,idx,color):
+    if b == '1':
+        uh.set_pixel(x,idx,color)
+    sleep(1.0/FPS)
+
+while True:
+    now = datetime.datetime.now()
+    month = bin(now.month)[2:][::-1]
+    day = bin(now.day)[2:][::-1]
+    hour = bin(now.hour)[2:][::-1]
+    minute = bin(now.minute)[2:][::-1]
+    second = bin(now.second)[2:][::-1]
+    uh.clear()
+    for idx,b in enumerate(hour):
```

Comment:
Since it's repeated, you should move this for loop into your `draw_time` function.

In fact you should probably move your `bin(number)[2:][::-1]` into there too since that's also repeated once for each time.

That way you'd just be passing in the raw time number, the vertical offset and the colour and the conversion and display (both closely related) will happen in one tidy place..

Reply:
I'm have already started this last night.